content/blogposts/2023/planes-are-not-faster-cars.md

```diff
@@ -0,0 +1,108 @@
+
+
+<table>
+  <tr>
+   <td>Drive to NYC</td>
+   <td>Fly to NYC</td>
+   <td>Time saved</td>
+  </tr>
+  <tr>
+   <td>43 hours</td>
+   <td>5 hours</td>
+   <td>38 hours</td>
+  </tr>
+</table>
+
+
+(I’m leaving out setup/teardown for flying, and also stopping for food/gas/sleep/etc while driving, but I already said this was the wrong model, so I don’t care how accurate it is.)
+
+Instead, this is how to think about it:
+
+<table>
+  <tr>
+   <td>Drive to NYC</td>
+   <td>Fly to NYC</td>
+   <td>Time saved</td>
+  </tr>
+  <tr>
+   <td>NOPE</td>
+   <td>5 hours</td>
+   <td>???</td>
+  </tr>
+</table>
+
+
+Infinity hours? Or zero? Arguably, flying actually _costs_ time compared to driving because I just wouldn’t drive that distance at all (so zero hours spent). But time saved is missing the point. The point is that _I actually went to NYC_.
+
+
+> **If the time-cost is so large that you’re never going to pay it, time-saving improvements are actually about making the effectively-impossible become possible.**
+
+Now back to those setup/teardown costs I mentioned. Let’s set the dial back to my early career in the (cough) late 1900s. The only programming language I really knew was C/C++. But Ruby was just hitting North America, and for me it was love at first sight.
+
+It was just so _easy!_ I could get the same thing done in roughly 1/10th the number of lines of code! (Yes, it ran 100x slower, but even 1900s computers were so fast that it often didn’t matter.) Very impressive time-savings!
+
+But that wasn’t the real magic. The real magic was that, without the setup cost of creating a new project (makefiles or project files, compiler settings, where is this going to live, etc), a whole new realm of _simple_ problems opened up!
+
+In Ruby, you don’t even need to open your text editor if the program is small enough; just pass the code into the interpreter on the command line. Like I would literally _never_ write a C program to tell me what time it will be in a million seconds. In Ruby:
+
+
+```
+➤  ruby -e 'p Time.now; p Time.now + 1e6;'
+2022-11-09 12:33:07.18994 -0800
+2022-11-21 02:19:47.189964 -0800
+```
+
+
+I thought of a question I could answer with code, wrote the code, got the answer — in just a few _seconds_.
+
+How much time did I save compared to writing this in C? It’s irrelevant: I would never have bothered writing something so trivial in C. The magic of Ruby wasn’t in the time saved. The magic was in the parts of my world that opened up. Suddenly, I could go from question-to-code-to-answer in just a few seconds!
+
+Both for the very large things (travel to distance cities) and for the very small things (simple questions I have where the setup cost just wasn’t going to be worth it), the principle is the same:
```

Comment:
Minor edit, I think this is meant to be "distant" rather than "distance":
```suggestion
Both for the very large things (travel to distant cities) and for the very small things (simple questions I have where the setup cost just wasn’t going to be worth it), the principle is the same:
```